frontend/src/metabase/hoc/Favicon.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { useEffect } from "react";
+import _ from "underscore";
+import MetabaseSettings from "../lib/settings";
+
+const DEFAULT_FAVICON = () => MetabaseSettings.get("application-favicon-url");
+export const LOAD_COMPLETE_FAVICON = "app/assets/img/blue_check.png";
+
+const resolveFavicon = (setterOrGetter, props) => {
+  if (typeof setterOrGetter === "string") {
+    return setterOrGetter;
+  } else if (typeof setterOrGetter === "function") {
+    const result = setterOrGetter(props);
+    if (result == null) {
+      return DEFAULT_FAVICON();
+    } else if (result instanceof String || typeof result === "string") {
+      return result;
+    }
+  }
+};
+
+const withFavicon = faviconSetterOrGetter => ComposedComponent => {
+  const WithFavicon = props => {
+    const favicon = resolveFavicon(faviconSetterOrGetter, props);
+
+    useEffect(() => {
+      document.querySelector('link[rel="icon"]').setAttribute("href", favicon);
```

Comment:
Do we need to a return a cleanup function here? i.e. to set the default favicon in case the component unmounts

Reply:
Just added :)